query/src/org/labkey/query/QueryWriter.java

```diff
@@ -68,6 +70,13 @@ public void write(Container object, ImportContext<FolderDocument.Folder> ctx, Vi
         List<QueryDefinition> queries = new ArrayList<>(QueryServiceImpl.get().getQueryDefsAndMetadataOverrides(ctx.getUser(), c));
         FileNameUniquifier fileNameUniquifier = new FileNameUniquifier();
 
+        if (ctx.getClass().equals(FolderExportContext.class))
+        {
+            Set<String> queryKeysToExport = ((FolderExportContext)ctx).getQueryKeys();
+            if (queryKeysToExport != null)
+                queries.removeIf(queryDef -> !queryKeysToExport.contains(queryDef.getQueryKey()));
+        }

```

Comment:
Why exclude `FolderExportContext` subclasses? Is there a chance they would want this behavior?

```suggestion
        if (ctx instanceof FolderExportContext feCtx)
        {
            Set<String> queryKeysToExport = feCtx.getQueryKeys();
            if (queryKeysToExport != null)
                queries.removeIf(queryDef -> !queryKeysToExport.contains(queryDef.getQueryKey()));
        }

```

Reply:
Good point. I was copying over logic from the ReportWriter. I've updated both to your suggestion. Thanks.